ppl/lake/walk.go

```diff
@@ -111,32 +114,40 @@ func tsDirEntriesToChunks(ctx context.Context, lk *Lake, filterSpan nano.Span, t
 			m[id] = s
 		}
 	}
-	var chunks []chunk.Chunk
+
+	var mu sync.Mutex
+	chunks := make([]chunk.Chunk, 0, len(m))
+	group, ctx := errgroup.WithContext(ctx)
+
 	for id, seen := range m {
 		if !seen.meta || !seen.data {
 			continue
 		}
-		dir := tsDir.path(lk)
-		mdPath := chunk.MetadataPath(dir, id)
-		b, err := lk.immfiles.ReadFile(ctx, mdPath)
-		if err != nil {
-			return nil, err
-		}
+		id := id
+		group.Go(func() error {
+			dir := tsDir.path(lk)
+			mdPath := chunk.MetadataPath(dir, id)
+			b, err := lk.immfiles.ReadFile(ctx, mdPath)
+			if err != nil {
+				return err
+			}
 
-		md, err := chunk.UnmarshalMetadata(b, lk.DataOrder)
-		if err != nil {
-			if zqe.IsNotFound(err) {
-				continue
+			md, err := chunk.UnmarshalMetadata(b, lk.DataOrder)
+			if err != nil {
+				if zqe.IsNotFound(err) {
```

Comment:
This 'is not found' check looks funny right here. I know that's not the intent of this PR, but I think it's supposed to happen after the ReadFile?

Reply:
Yep! Will Fix.